### PR TITLE
improved evaluation script + small mask fix

### PR DESCRIPTION
**Pull request by danielmend**

### CODE REVIEW

File: src/aggregation/aggregator_wrapper.py [hunk 1]
```diff
@@ -11,15 +11,15 @@ def __init__(self, clip):
         self.transformer = clip.transformer
         self.ln_final = clip.ln_final
         self.text_projection = clip.text_projection
-        self.attn_mask = clip.attn_mask.cuda()
-        # device = self.token_embedding.device
-        # self.attn_mask = self.attn_mask.cuda()
+        self.attn_mask = clip.attn_mask
 
     def forward(self, text):
+        device = next(self.transformer.parameters()).device
```
Comment (iejMac) [inline src/aggregation/aggregator_wrapper.py:17]: why do we need this?

File: src/aggregation/self_attention_pool.py [hunk 2]
```diff
@@ -104,8 +104,11 @@ def forward(self, x, attn_masks=None):
         for attn, ff in self.layers:
             x = attn(x, attn_mask=attn_masks) + x
             x = ff(x) + x
-
-        x = torch.mean(x, 1)
+        if attn_masks is not None: # masked mean
+            denom = torch.sum(attn_masks, dim=-1, keepdim=True)
```
Comment (iejMac) [inline src/aggregation/self_attention_pool.py:108]: do we need this? is this tested?

File: src/evaluation/retrieval.py [hunk 3]
```diff
@@ -3,27 +3,28 @@
...
 
-def retrieval_evaluation(model_video, data, multicaption=False):
+def retrieval_evaluation(model_video, data, multicaption=False, device="cuda"):
     if type(data) == dict:
         dataloader = data["val"].dataloader
     else:
         dataloader = data
     model_video.eval()
-    device = "cuda" if torch.cuda.is_available() else "cpu"
+    model_video.to(device, non_blocking=True)
```
Comment (iejMac) [inline src/evaluation/retrieval.py:12]: this has nothing to do with retrieval_evaluation, should be done outside the script

File: src/aggregation/self_attention_pool.py (hunk 2 above)
Comment (danielmend) [reply src/aggregation/self_attention_pool.py:108]: it worked to take the spikes out of the training loss curve and i tested the masked mean part locally, but i can do some further testing on this

File: src/aggregation/aggregator_wrapper.py (hunk 1 above)
Comment (danielmend) [reply src/aggregation/aggregator_wrapper.py:17]: the clip.attn_mask is not a parameter of the VideoCLIP module and so when the model_video.to(device) is called the attn_mask stays on CPU, so this was just a hack to get the device of the aggregator and manually switch the clip.attn_mask to the correct device
